### DISCUSSION TITLE
How To Answer The Engineers Questions?

### DISCUSSION THREAD

**Discussion by danielgasser** [AntonOsika/gpt-engineer#514] [Q&A] [answered]
I'm taking the snake example as a reference here. When starting the engineer on a new project, we get these:

```
Areas that need clarification:
1. What specific components need to be split into separate files?
2. What is the desired functionality of the keyboard control?
3. Are there any specific design patterns or frameworks that need to be followed for the MVC implementation?
4. Are there any specific requirements or constraints for the snake game implementation?
5. Is there a specific version of Python that needs to be used for this project?
``` 
My way of answering is by repeating the question and responding on a new line, like this:
``` 
2. What is the desired functionality of the keyboard control?
We use the arrow keys to move the snake around and the ESC key to exit the game
``` 
If I answer one question like the above and press ENTER, the engineer states `Nothing more to clarify.` and starts building. What about the other questions?

On a more complex project, the questions get obviously more complex as well as the answers.

If I answer all the questions at once by repeating the question on a separate line and responding on the next line, it gets messy because the engineer starts asking further questions like 

``` 
2.1. ...?
2.2. ...?
2.3. ...?
``` 
but it doesn't seem to take all my answers into consideration.

Here's my question.
#### Awesome thing you built here @AntonOsika! Thanks for that

**Comment by adski27** [comment, marked as answer]
Not sure if there's a better way - but I just type all on 'one line' with no CR/Enter. I'll preface each answer with the number.
So rather than:
1) Yes please add that suggestion
2) Needs to be 3

It would be:
1) Yes please add that suggestion 2) Needs to be 3
